scripts/update_KSJCodeTranslations.R

```diff
@@ -0,0 +1,298 @@
+  if (file.exists(destfile)) return(FALSE)
+
+  curl::curl_download(url, destfile)
+  Sys.sleep(1)
+  return(TRUE)
+})
+
+list_codelist_urls <- function(x) {
+  read_html(x, encoding = "CP932") %>%
+    html_nodes(css = "a") %>%
+    html_attr("href") %>%
+    stringr::str_subset("codelist")
+}
+
+
+### datalist ###
+
+datalist_urls <- KSJIdentifierDescriptionURL$url
+datalist_destfiles <- file.path("downloaded_html", paste0("datalist-", basename(datalist_urls)))
+
+result <- purrr::map2(datalist_urls,
+                      datalist_destfiles,
+                      download_safely)
+
+if (any(purrr::map_lgl(result, ~ !is.null(.$error)))) {
+  stop("some error happened")
+}
+
+### codelist ###
+
+codelist_urls <- purrr::map(datalist_destfiles, list_codelist_urls) %>%
+  purrr::flatten_chr() %>%
+  unique %>%
+  sort
+
+codelist_destfiles <- file.path("downloaded_html", paste0("codelist-", basename(codelist_urls)))
+
+result <- purrr::map2(codelist_urls,
+                      codelist_destfiles,
+                      download_safely)
+
+if (any(purrr::map_lgl(result, ~ !is.null(.$error)))) {
+  stop("some error happened")
+}
+
+
+# extract tables --------------------------------------------------
+
+### define functions ###
+
+extract_tables_html <- function(html) {
+  read_html(html, encoding = "CP932") %>%
+    html_nodes(css = "table table") %>%
+    keep(~ stringr::str_detect(as.character(.), "属性情報|地物情報"))
+}
+
+extract_tables_html_safely <- purrr::safely(extract_tables_html)
+
+### process HTMLs ###
+
+datalist_destfiles <- list.files("downloaded_html", pattern = "datalist-.*", full.names = TRUE)
+list_of_tables_html_wrapped <- purrr::map(datalist_destfiles, extract_tables_html_safely)
+# check errors
+datalist_destfiles[map_lgl(list_of_tables_html_wrapped, ~ !is.null(.$error))]
+
+list_of_tables_html <- purrr::map(list_of_tables_html_wrapped, "result") %>%
+  rlang::set_names(KSJIdentifierDescriptionURL$identifier)
+
+
+# Re-split tables -----------------------------------------------------
+
+### define functions ###
+
+resplit_tables_html <- function(tables) {
+  map(tables, resplit_one_table_html) %>%
+    flatten
+}
+
+resplit_one_table_html <- function(table) {
+  tr_list_raw <- html_children(table)
+
+  # if tr or first td have bgcolor attribute, the rows bellow belongs to new group.
+  has_bgcolor_tr  <- map_lgl(tr_list_raw, ~ !is.na(html_attr(., "bgcolor")))
+
+  has_bgcolor_tds <- tr_list_raw %>%
+    map(html_nodes, "td") %>%
+    map(map_chr, html_attr, "bgcolor") %>%
+    map(~ !is.na(.))
+  has_bgcolor_first_td <- map_lgl(has_bgcolor_tds, first)
+  has_bgcolor_all_td   <- map_lgl(has_bgcolor_tds, all)
+
+  is_start_of_different_table <- has_bgcolor_tr | has_bgcolor_first_td
+
+  table_id <- cumsum(is_start_of_different_table)
+  # if all td has bgcolor, it is a header
+  has_header <- (has_bgcolor_tr | has_bgcolor_all_td)[is_start_of_different_table]
+  # if a table has multiple rows, first td has a rowspan attribute
```

Comment:
This assumption is wrong; we have to subset by rowspan before splitting tables, because there are some weird ones whose headers don't have rowspans like this:

http://nlftp.mlit.go.jp/ksj/gml/datalist/KsjTmplt-P33.html